src/theory/strings/rewriter_str.h

```diff
@@ -0,0 +1,88 @@
+/*********************                                                        */
+/*! \file rewriter_str.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds, Andres Noetzli, Tianyi Liang
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2019 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Rewrite rules for string-specific operators in theory of strings
+ **
+ **/
+
+#include "cvc4_private.h"
+
+#ifndef CVC4__THEORY__STRINGS__REWRITER_STR_H
+#define CVC4__THEORY__STRINGS__REWRITER_STR_H
+
+#include "expr/node.h"
+#include "theory/strings/theory_strings_rewriter.h"
+
+namespace CVC4 {
+namespace theory {
+namespace strings {
+
+/**
+ * An extension of TheoryStringsRewriter that handles operators that
+ * are specific to strings (and cannot be applied to sequences).
+ */
+class RewriterStr : public TheoryStringsRewriter
```

Comment:
How about calling them `TheorySequenceRewriter` (for what is currently called `TheoryStringsRewriter`) and `TheoryStringsRewriter` (for what is currently `RewriterStr`)?

Reply:
Sure, I can update this.